### PR TITLE
Add benchmark tool to test chain performance

### PR DESCRIPTION
**Pull request by RogerKSI**
## Implementation details
- add benchmark tool to test chain performance

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [x] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: benchmark/helper_test.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+			DataSourceId: dataSourceId,
+			Scenario:     scenario,
+			Value:        value,
+		}),
+		AskCount:   1,
+		MinCount:   1,
+		ClientID:   "",
+		FeeLimit:   sdk.Coins{sdk.NewInt64Coin("uband", 1)},
+		PrepareGas: GasLimit,
+		ExecuteGas: GasLimit,
```
Comment (colmazia) [inline benchmark/helper_test.go:51]: *Optional
maybe consider separate to PrepareGasLimit and ExecuteGasLimit for better understanding and more customization in the future

File: benchmark/helper_test.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	numTxs int,
+) []sdk.Tx {
+	txs := make([]sdk.Tx, numTxs)
+
+	for i := 0; i < numTxs; i++ {
+		txs[i], _ = testapp.GenTx(
+			txConfig,
+			msgs,
+			sdk.Coins{sdk.NewInt64Coin("uband", 1)},
+			1000000000,
```
Comment (colmazia) [inline benchmark/helper_test.go:128]: why 1000000000

File: testing/testapp/setup.go [hunk 3]
```diff
@@ -91,7 +91,7 @@ const (
 var DefaultConsensusParams = &abci.ConsensusParams{
 	Block: &abci.BlockParams{
 		MaxBytes: 200000,
-		MaxGas:   2000000,
+		MaxGas:   -1,
```
Comment (colmazia) [inline testing/testapp/setup.go:94]: why set maxgas to -1

Comment (RogerKSI) [reply testing/testapp/setup.go:94]: -1 means no limit so that we can test with many txs in one block.

File: benchmark/helper_test.go (hunk 1 above)
Comment (RogerKSI) [reply benchmark/helper_test.go:51]: ok. let me separate the variable for prepare and execute.

File: benchmark/helper_test.go (hunk 2 above)
Comment (RogerKSI) [reply benchmark/helper_test.go:128]: I'll update to use maxint.

**Review by colmazia — APPROVED** [review]

File: benchmark/app_test.go [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+	_, res, err := ba.CallDeliver(tx)
+	require.NoError(ba.TB, err)
+
+	evt := res.Events[len(res.Events)-1]
+	attr := evt.Attributes[0]
+	did, err := strconv.Atoi(string(attr.Value))
+	require.NoError(ba.TB, err)
+
+	return did
+}
```
Comment (Kissadada) [inline benchmark/app_test.go:101]: these two functions look very similar, can we refactor them?

File: benchmark/bench_test.go [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+							int64(types.DefaultMaxCalldataSize),
+							env,
+						)
+						b.StopTimer()
+					}
+				})
+			}
+		}
+	}
+}
```
Comment (Kissadada) [inline benchmark/bench_test.go:165]: ditto

File: benchmark/app_test.go (hunk 4 above)
Comment (RogerKSI) [reply benchmark/app_test.go:101]: okay, already created new two functions (one for delivering msg, second for get value for first event)

File: benchmark/bench_test.go (hunk 5 above)
Comment (RogerKSI) [reply benchmark/bench_test.go]: done, create setup function for owasm env.

**Review by satawatnack — APPROVED** [review]

**Review by satawatnack — APPROVED** [review]

**Pull request merged**